### PR TITLE
feat: integrate dodoex aggregator

### PR DESCRIPTION
**Pull request by 0xKoaj**

### CODE REVIEW

File: src/services/quotes/quote-sources/dodo-quote-source.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
+import { Chains } from '@chains';
+import { AlwaysValidConfigAndContextSource } from './base/always-valid-source';
+import { QuoteParams, QuoteSourceMetadata, SourceQuoteResponse } from './types';
+import qs from 'qs';
+import { addQuoteSlippage, calculateAllowanceTarget, failed } from './utils';
+import { parseUnits } from 'viem';
+
+const DODO_DEX_METADATA: QuoteSourceMetadata<DodoDexSupport> = {
+  name: 'DODO Dex',
```
Comment (0xsambugs) [inline src/services/quotes/quote-sources/dodo-quote-source.ts:9]: I think we can go with DODO
```suggestion
  name: 'DODO',
```

File: src/services/quotes/quote-sources/dodo-quote-source.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      Chains.HECO.chainId,
+      Chains.LINEA.chainId,
+      Chains.MOONRIVER.chainId,
+      Chains.OKC.chainId,
+      Chains.OPTIMISM.chainId,
+    ],
+    swapAndTransfer: false,
+    buyOrders: false,
+  },
+  logoURI: '',
```
Comment (0xsambugs) [inline src/services/quotes/quote-sources/dodo-quote-source.ts:30]: I've asked Fibo por a logo IPFS hash

File: src/services/quotes/quote-sources/dodo-quote-source.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+};
+type DodoDexConfig = { apiKey: string };
+type DodoDexSupport = { buyOrders: false; swapAndTransfer: false };
+export class DodoDexQuoteSource extends AlwaysValidConfigAndContextSource<DodoDexSupport, DodoDexConfig> {
```
Comment (0xsambugs) [inline src/services/quotes/quote-sources/dodo-quote-source.ts:34]: Is the `apiKey` necessary, or optional? If it is necessary, then it wouldn't be always valid

File: src/services/quotes/quote-sources/dodo-quote-source.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    const buyAmount = parseUnits(quoteResult.data.resAmount.toString(), quoteResult.data.targetDecimals);
+    const { targetApproveAddr, data, to, value } = quoteResult.data;
+    const quote = {
+      sellAmount: order.sellAmount,
+      buyAmount,
+      estimatedGas: undefined,
+      allowanceTarget: targetApproveAddr ?? calculateAllowanceTarget(sellToken, to),
+      tx: {
+        calldata: data,
+        to: to,
```
Comment (0xsambugs) [inline src/services/quotes/quote-sources/dodo-quote-source.ts:83]: Small nit
```suggestion
        to,
```

File: src/services/quotes/quote-sources/dodo-quote-source.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    if (quoteResult.status < 0) {
+      failed(DODO_DEX_METADATA, chain, sellToken, buyToken, quoteResult.data);
+    }
+    const buyAmount = parseUnits(quoteResult.data.resAmount.toString(), quoteResult.data.targetDecimals);
+    const { targetApproveAddr, data, to, value } = quoteResult.data;
+    const quote = {
+      sellAmount: order.sellAmount,
+      buyAmount,
+      estimatedGas: undefined,
+      allowanceTarget: targetApproveAddr ?? calculateAllowanceTarget(sellToken, to),
```
Comment (0xsambugs) [inline src/services/quotes/quote-sources/dodo-quote-source.ts:80]: When can be `targetApproveAddr` be undefined? Just curious

**Review by 0xsambugs — COMMENTED** [review]
Looks good, just left some small comments & questions

File: src/services/quotes/quote-sources/dodo-quote-source.ts (hunk 5 above)
Comment (0xKoaj) [reply src/services/quotes/quote-sources/dodo-quote-source.ts:80]: theirs documentation says: "If the token being sold is a gas token such as ETH (on BNB or MATIC), this parameter will be empty."

File: src/services/quotes/quote-sources/dodo-quote-source.ts (hunk 2 above)
Comment (0xKoaj) [reply src/services/quotes/quote-sources/dodo-quote-source.ts:30]: Excellent, thank you! 🚀

File: src/services/quotes/quote-sources/dodo-quote-source.ts (hunk 5 above)
Comment (0xsambugs) [reply src/services/quotes/quote-sources/dodo-quote-source.ts:80]: Then lets do `allowanceTarget: targetApproveAddr ?? Addresses.ZERO_ADDRESS`

**Review by 0xsambugs — APPROVED** [review]
Looks good!
Let's wait for the logo and then merge 🔥

**Pull request merged**